webapp/channels/src/components/channel_settings_modal/channel_settings_access_rules_tab.tsx

```diff
@@ -161,20 +161,47 @@ function ChannelSettingsAccessRulesTab({
         }));
     }, [formatMessage]);
 
+    // Helper function to detect empty rules state
+    const isEmptyRulesState = useMemo((): boolean => {
+        const hasChannelRules = expression && expression.trim().length > 0;
+        const hasSystemPolicies = systemPolicies && systemPolicies.length > 0;
+
+        // Edge case: No channel rules AND no system policies at all (applied or not)
+        return !hasChannelRules && !hasSystemPolicies;
+    }, [expression, systemPolicies]);
+
+    // Auto-disable auto-sync when entering empty rules state (but only if not forced by system policy)
+    useEffect(() => {
+        if (isEmptyRulesState && autoSyncMembers && !systemPolicyForcesAutoSync) {
+            setAutoSyncMembers(false);
```

Comment:
We have multiple `useEffects` calling `setAutoSyncMembers`, as such, it may get confusing to figure out when and how the value is set from and why, especially when debugging some bug around this code. As such, I'd say we should refactor this to use a single `useEffect` or something else to set the value and serve as a simple, straight forward, single source of truth on how the value is set.

Reply:
thanks for the feedback. I unified the effects in the last commit.